app/helpers/common_parameters_helper.rb

```diff
@@ -40,7 +40,8 @@ def use_puppet_default_help link_title = nil, title = _("Use Puppet default")
   end
 
   def hidden_value_field(f, field, value, disabled, options = {})
-    if f.object.hidden_value?
+    hidden = options.delete(:hidden_value)
+    if f.object.hidden_value? || hidden
```

Comment:
I would switch the order of this `||` in case the optin is passed explicitly to save checking the form object.